src/workerd/api/node/tests/zlib-nodejs-test.js

```diff
@@ -1054,33 +1052,473 @@ export const zlibInvalidInput = {
+      unzip.write(Buffer.concat([abcEncoded, defEncoded.slice(0, offset)]));
+
+      // Write remaining bytes of "def"
+      unzip.end(defEncoded.slice(offset));
+
+      await promise;
+    }
+  },
+};
+
+// Test taken from:
+// https://github.com/nodejs/node/blob/fc02b88f89f8d5abf5ee4414a1026444c18d77b3/test/parallel/test-zlib-not-string-or-buffer.js
+export const zlibNotStringOrBuffer = {
+  test() {
+    [undefined, null, true, false, 0, 1, [1, 2, 3], { foo: 'bar' }].forEach(
+      (input) => {
+        assert.throws(
+          () => zlib.deflateSync(input),
+          // FIXME: our errors don't have the same code as in NodeJS
+          {
+            name: 'TypeError',
+          }
+        );
+      }
+    );
+  },
+};
+
+// Test taken from:
+// https://github.com/nodejs/node/blob/fc02b88f89f8d5abf5ee4414a1026444c18d77b3/test/parallel/test-zlib-from-gzip-with-trailing-garbage.js
+export const zlibFromGzipWithTrailingGarbage = {
+  async test() {
+    // Should ignore trailing null-bytes
+    let data = Buffer.concat([
+      zlib.gzipSync('abc'),
+      zlib.gzipSync('def'),
+      Buffer.alloc(10),
+    ]);
+
+    assert.strictEqual(zlib.gunzipSync(data).toString(), 'abcdef');
+
+    {
+      const { promise, resolve } = Promise.withResolvers();
+
+      zlib.gunzip(data, (err, result) => {
+        assert.ifError(err);
+        assert.strictEqual(
+          result.toString(),
+          'abcdef',
+          `result '${result.toString()}' should match original string`
+        );
+        resolve();
+      });
+
+      await promise;
+    }
+
+    // If the trailing garbage happens to look like a gzip header, it should
+    // throw an error.
+    data = Buffer.concat([
+      zlib.gzipSync('abc'),
+      zlib.gzipSync('def'),
+      Buffer.from([0x1f, 0x8b, 0xff, 0xff]),
+      Buffer.alloc(10),
+    ]);
+
+    assert.throws(
+      () => zlib.gunzipSync(data),
+      /^Error: unknown compression method$/
+    );
+
+    {
+      const { promise, resolve } = Promise.withResolvers();
+      zlib.gunzip(data, (err, result) => {
+        assert(err instanceof Error);
+        // FIXME: our messages don't have code
+        // FIXME: Do our messages have a redundant "Error: " word?
+        assert.match(err.message, /unknown compression method/);
+        assert.strictEqual(result, undefined);
+        resolve();
+      });
+
+      await promise;
+    }
+
+    // In this case the trailing junk is too short to be a gzip segment
+    // So we ignore it and decompression succeeds.
+    data = Buffer.concat([
+      zlib.gzipSync('abc'),
+      zlib.gzipSync('def'),
+      Buffer.from([0x1f, 0x8b, 0xff, 0xff]),
+    ]);
+
+    assert.throws(
+      () => zlib.gunzipSync(data),
+      /^Error: unknown compression method$/
+    );
+
+    {
+      const { promise, resolve } = Promise.withResolvers();
+      zlib.gunzip(data, (err, result) => {
+        assert(err instanceof Error);
+        // FIXME: our messages don't have code
```

Comment:
Can you replace all FIXME's with TODO(soon)?

```suggestion
        // TODO(soon): our messages don't have code
```